File: 240808/상품 코드/product-code.cpp
#include <iostream>
#include <string>

using namespace std;

class My {
public:
    string name;
    int code;

    // 생성자를 public으로 선언 및 기본값으로 초기화
    My(string name = "codetree", int code = 50) {
        this->name = name;
        this->code = code;
    }
};

int main() {
    string name;
    int code;

    // 기본 생성자 사용하여 객체 초기화
    My a;

    // 기본 생성자 초기화된 객체 정보 출력
    cout << "product " << a.code << " is " << a.name << '\n';

    // 사용자 입력 받기
    cin >> name >> code;
    
    // 사용자 입력값으로 객체 초기화
    a = My(name, code);

    // 업데이트된 객체 정보 출력
    cout << "product " << a.code << " is " << a.name << '\n';

    return 0;
}

File: 241019/상품 코드/product-code.py
class product :
    def __init__(self, name=0, code=0) :
        self.name = name
        self.code = code

p1 = product("codetree", 50)
p2 = product()

p2.name, p2.code = tuple(input().split())

print(f"product {p1.code} is {p1.name}")
print(f"product {p2.code} is {p2.name}")